lib/cocoapods-core/cdn_source.rb

```diff
@@ -60,13 +74,23 @@ def preheat_existing_files
       files_to_update = files_definitely_to_update + deprecated_local_podspecs - ['deprecated_podspecs.txt']
       debug "CDN: #{name} Going to update #{files_to_update.count} files"
 
-      concurrent_requests_catching_errors do
+      errors = []
+
+      Sync do
```

Comment:
```
Sync do |task|
  # ...
  task.async do
    # ...
```

is better.

Reply:
Use `barrier` for both single and multiple tasks here, thanks!